### PR TITLE
[TTAHUB-2367] Updates to FEI Root Cause

### PR DESCRIPTION
**Pull request by AdamAdHocTeam**
## Description of change

This change keeps FEI Root cause editable until the goal is closed. This is true for all paths that the root cause can be changed from (AR and RTR). It also adds a new root cause option called 'Unavailable'.

## How to test

- Review the code.

**Make sure each of the below is true:**
- add option to the root cause dropdown for "Unavailable" for Activity Report form
- add option to the root cause dropdown for "Unavailable" for RTR > RTTAPA > goal creation/edit
- add "Unavailable" for the root cause filter on regional dashboard, activity report and resource dashboard page
- root cause is editable (delete existing root causes or add up to two root causes) on the RTR at any time
- root cause is editable (delete existing root causes or add up to two root causes) on the AR
           - using the existing rules. If one grant number is on AR this can happen.

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-2367

### Every PR

- [x] Meets issue criteria
- [x] JIRA ticket status updated
- [x] Code is meaningfully tested
- [x] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [ ] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions
- [ ] UI review complete

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [x] Update JIRA ticket status

### CODE REVIEW

File: frontend/src/pages/ActivityReport/Pages/components/GoalForm.js [hunk 1]
```diff
@@ -164,6 +164,7 @@ export default function GoalForm({
 
   const prompts = combinePrompts(templatePrompts, goal.prompts);
   const isCurated = goal.isCurated || false;
+  const canEditFei = status !== 'Closed';
```
Comment (AdamAdHocTeam) [inline frontend/src/pages/ActivityReport/Pages/components/GoalForm.js:167]: Not sure if its possible to have a closed goal on a open AR.

File: frontend/src/components/ConditionalFields.js [hunk 2]
```diff
@@ -68,18 +63,6 @@ export default function ConditionalFields({
...
-          if (current) {
-            return current.response;
-          }
-
-          return [];
-        })();
-
-        const completions = validationsAndCompletions.confirmResponseComplete(prompt.validations);
-        const isComplete = completions.every((completion) => completion(initialValue));
-
```
Comment (thewatermethod) [inline frontend/src/components/ConditionalFields.js:82]: love to see all the red

File: frontend/src/pages/ActivityReport/Pages/components/GoalForm.js (hunk 1 above)
Comment (thewatermethod) [reply frontend/src/pages/ActivityReport/Pages/components/GoalForm.js:167]: right now "goalsForGrants" excludes closed and suspended goals

File: frontend/src/pages/ActivityReport/Pages/components/__tests__/RecipientReviewSection.js [hunk 3]
```diff
@@ -91,4 +93,10 @@ describe('RecipientReviewSection', () => {
     expect(screen.getByText(/https:\/\/www.govtest2.com/)).toBeInTheDocument();
     expect(screen.getByText(/TTA Provided/)).toBeInTheDocument();
   });
+
+  it('renders without goalsAndObjectives', async () => {
+    console.log('\n\n\n-----Start of Test');
```
Comment (thewatermethod) [inline frontend/src/pages/ActivityReport/Pages/components/__tests__/RecipientReviewSection.js:98]: remove console

**Review by thewatermethod — APPROVED** [review]

File: frontend/src/pages/ActivityReport/Pages/components/GoalForm.js (hunk 1 above)
Comment (AdamAdHocTeam) [reply frontend/src/pages/ActivityReport/Pages/components/GoalForm.js:167]: Ok so from this screen we can hard code true, correct?

Comment (thewatermethod) [reply frontend/src/pages/ActivityReport/Pages/components/GoalForm.js:167]: think so

File: frontend/src/components/ConditionalFields.js (hunk 2 above)
Comment (AdamAdHocTeam) [reply frontend/src/components/ConditionalFields.js:82]: I don't have the lines in latest version did you pull?

Comment (thewatermethod) [reply frontend/src/components/ConditionalFields.js:82]: No idea what you mean. What I said was that I like to see that you've removed files.

File: frontend/src/pages/ActivityReport/Pages/components/__tests__/RecipientReviewSection.js (hunk 3 above)
Comment (AdamAdHocTeam) [reply frontend/src/pages/ActivityReport/Pages/components/__tests__/RecipientReviewSection.js]: done

File: frontend/src/components/ConditionalFields.js (hunk 2 above)
Comment (AdamAdHocTeam) [reply frontend/src/components/ConditionalFields.js:82]: I guess what did you mean by this comment? I though you are saying I have unused lines of code. But I don't see these lines of code in latest version.

Comment (thewatermethod) [reply frontend/src/components/ConditionalFields.js:82]: sorry for the confusion, I was just saying I was glad to see you could remove all this code, no action required on your part

**Pull request merged**